patches/node/fix_lazyload_fs_in_esm_loaders_to_apply_asar_patches.patch

```diff
@@ -0,0 +1,103 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Samuel Attard <marshallofsound@electronjs.org>
+Date: Tue, 7 Mar 2023 16:17:41 -0800
+Subject: fix: lazyload fs in esm loaders to apply asar patches
+
+Changes { foo } from fs to just "fs.foo" so that our patching of fs is applied to esm loaders
+
+diff --git a/lib/internal/modules/esm/load.js b/lib/internal/modules/esm/load.js
+index a01852a241f9b4a575360a8464d77e3f5c6a5770..e6d8faa01afa51e33a9450b578278170d7c5b6a6 100644
+--- a/lib/internal/modules/esm/load.js
++++ b/lib/internal/modules/esm/load.js
+@@ -20,7 +20,7 @@ const experimentalNetworkImports =
+ 
+ const { Buffer: { from: BufferFrom } } = require('buffer');
+ 
+-const { readFile: readFileAsync } = require('internal/fs/promises').exports;
++const fs = require('fs');
+ const { URL } = require('internal/url');
+ const {
+   ERR_INVALID_URL,
+@@ -34,7 +34,7 @@ async function getSource(url, context) {
+   let responseURL = url;
+   let source;
+   if (parsed.protocol === 'file:') {
+-    source = await readFileAsync(parsed);
++    source = await fs.promises.readFile(parsed);
+   } else if (parsed.protocol === 'data:') {
+     const match = RegExpPrototypeExec(DATA_URL_PATTERN, parsed.pathname);
+     if (!match) {
+diff --git a/lib/internal/modules/esm/resolve.js b/lib/internal/modules/esm/resolve.js
+index 15415e0652864d059390bc7383e073c86f5d0c03..c8fd2ca438eb546f2149f9636d7454e81d44d5f9 100644
+--- a/lib/internal/modules/esm/resolve.js
++++ b/lib/internal/modules/esm/resolve.js
+@@ -27,11 +27,7 @@ const {
+ } = primordials;
+ const internalFS = require('internal/fs/utils');
+ const { BuiltinModule } = require('internal/bootstrap/loaders');
+-const {
+-  realpathSync,
+-  statSync,
+-  Stats,
+-} = require('fs');
```

Comment:
how come this style breaks with our asar patches? seems like it ought to work, right..?

Reply:
We set `fs.exists = ourExists`, because node runs **before** our patch if they destructure the fs module they capture the old `exists` and never use our patched one.